test/test_datapipe.py

```diff
@@ -1231,14 +1246,24 @@ def _helper(ref_fn, fn, input_col=None, output_col=None, error=None):
                     self.assertEqual(list(res_dp), list(ref_dp))
                     # Reset
                     self.assertEqual(list(res_dp), list(ref_dp))
+        _helper(lambda data: data, fn_n1_def, 0, 1)
+        _helper(lambda data: data, p_fn_n1, 0, 1)
```

Comment:
Add `_helper(None, p_fn_n1, (0, 1), error=ValueError)`

Reply:
This one throws `TypeError: fn_n1() got multiple values for argument 'd1'`

Reply:
Then, it seems `validate_input_col` wasn't able to catch such Error.

Reply:
Will update checker to check for partial functions correctly.